SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp

```diff
@@ -64,11 +65,19 @@ TArray<FWorkerComponentData> EntityFactory::CreateEntityComponents(USpatialActor
 	}
 
 	const FClassInfo& Info = ClassInfoManager->GetOrCreateClassInfoByClass(Class);
-	WorkerAttributeSet WorkerAttributeOrSpecificWorker{ Info.WorkerType.ToString() };
+
+	FName EffectiveWorkerType = Info.WorkerType;
+	const USpatialGDKSettings* SpatialSettings = GetDefault<USpatialGDKSettings>();
+
+	if (SpatialSettings->bEnableOffloading)
+	{
+		EffectiveWorkerType = ActorGroupManager->GetWorkerTypeForActorGroup(USpatialStatics::GetActorGroupForActor(Actor));
+	}
```

Comment:
Personal pref, change if you like:

```suggestion
        const USpatialGDKSettings* SpatialSettings = GetDefault<USpatialGDKSettings>();

	const FName EffectiveWorkerType = SpatialSettings->bEnableOffloading ? 
	         ActorGroupManager->GetWorkerTypeForActorGroup(USpatialStatics::GetActorGroupForActor(Actor)) :
	         Info.WorkerType;
```

Reply:
also prefer `EffectiveWorkerType` -> `AclAuthoritativeWorkerAttribute`

Reply:
I like it, changed AclAuthoritativeWorkerAttribute to AclAuthoritativeWorkerType, as it's just a string at this point.

Reply:
well, an FName, but you know what I mean